workspaces/ballerina/ballerina-extension/src/features/test-explorer/runner.ts

```diff
@@ -21,12 +21,69 @@ import { exec } from 'child_process';
 import { CancellationToken, TestRunRequest, TestMessage, TestRun, TestItem, debug, Uri, WorkspaceFolder, DebugConfiguration, workspace, TestRunProfileKind } from 'vscode';
 import { testController } from './activator';
 import { StateMachine } from "../../stateMachine";
-import { isTestFunctionItem, isTestGroupItem } from './discover';
+import { isTestFunctionItem, isTestGroupItem, isProjectGroupItem } from './discover';
 import { extension } from '../../BalExtensionContext';
 import { constructDebugConfig } from "../debugger";
 const fs = require('fs');
 import path from 'path';
 
+/**
+ * Extract project path from a test item
+ * Test IDs have the format: test:${projectPath}:${fileName}:${functionName}
+ */
+function getProjectPathFromTestItem(test: TestItem): string | undefined {
+    if (isTestFunctionItem(test)) {
+        // Extract from test ID: test:${projectPath}:${fileName}:${functionName}
+        const parts = test.id.split(':');
```

Comment:
if `projectPath` is absolute path, this logic will break on Windows.

on Windows project path contains `:` (colons) (e.g., `C:\Users\...\project`), which breaks the parsing logic in `getProjectPathFromTestItem()` and `belongsToFile()`. 

Change the separator to something that won't conflict with file paths, such as `|` (pipe): test|${projectPath}|${fileName}|${functionName}.